waitress/server.py

```diff
@@ -201,7 +201,7 @@ def get_server_name(self, ip):
             if c != '.' and not c.isdigit():
                 return server_name
         try:
-            if server_name == '0.0.0.0' or server_name == '::':
+            if server_name == '0.0.0.0' or server_name == '::' or server_name == '127.0.0.1' :
```

Comment:
Needs to update the existing [`test_get_server_name_with_ip`](https://github.com/Pylons/waitress/blob/5d84524635a1534953d380b2511ccaa1f0b1bee9/waitress/tests/test_server.py#L73) to use a different IP address, and add another test similar to [`test_get_server_name_0000`](https://github.com/Pylons/waitress/blob/5d84524635a1534953d380b2511ccaa1f0b1bee9/waitress/tests/test_server.py#L83).  Looks like we need one of those for `'::'`, too.

Reply:
Also, once the set of compared values gets this long, I'd switch to declaring a module-scope `_LOCALHOST_IPS` list and test for containment.  The check doesn't need to be inside the `try: ... except:` block either.  E.g.:


```python
_LOCALHOST_IP_ADDRESSES = [
    '0.0.0.0',
    '::',
    '127.0.0.1',
]
...
        for c in server_name:
            if c != '.' and not c.isdigit():
                return server_name
        if server_name in _LOCALHOST_IP_ADRESSES:
            return 'localhost'
        try:
            server_name = self.socketmod.gethostbyaddr(server_name)[0]
        except socket.error: # pragma: no cover
             pass
        return server_name
```

Reply:
This code needs to be refactored in general. Trying to guess the server name from the IP address is wrong, since 0.0.0.0 is not "localhost", it means bind to all IP's on this system, and server_name should be set to whatever the rDNS is for the IP the remote user connected to.

Reply:
(Same for `::`)